service/history/nDCActivityReplicator.go

```diff
@@ -215,84 +192,132 @@ func (r *nDCActivityReplicatorImpl) SyncActivity(
 	)
 }
 
-func (r *nDCActivityReplicatorImpl) shouldApplySyncActivity(
+func (r *nDCActivityReplicatorImpl) refreshTask(
```

Comment:
Name seems to suggest you are doing some action `refreshTask` but code seems to be performing a check whether task should be refreshed or not.  Might want to rename this for code readability.